src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -1538,7 +1538,7 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 		}
 		const terminalKey = terminal.instanceId.toString();
 		const terminalData = { terminal: terminal, lastTask: taskKey, group };
-		terminal.onDisposed(() => this._deleteTaskAndTerminal(terminal, terminalData));
+		this._register(terminal.onDisposed(() => this._deleteTaskAndTerminal(terminal, terminalData)));
```

Comment:
This listener should be tracked and disposed of after it's been disposed of. I think this would work?

```ts
const onDisposedListener = this.register(terminal.onDisposed(() => {
  this._deleteTaskAndTerminal(terminal, terminalData));
  this.delete(onDisposedListener);
});
```